packages/snap/src/rpc/generateTransactionPayload.ts

```diff
@@ -20,8 +20,12 @@ export async function generateTransactionPayload(
     }),
     nonce
   };
+
   // define transaction method
-  const transaction: SubmittableExtrinsic<'promise'> = api.tx.balances.transfer(to, amount);
+  const transaction: SubmittableExtrinsic<'promise'> = api.tx.balances.transferAllowDeath(
```

Comment:
Why not use transferKeepAlive?

Reply:
agreed, it should be `transferKeepAlive` which is more safe for users.

Reply:
yeah, probably indeed transferKeepAlive is safer